hack/run-job-aggragator-analyzor.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+## Run this script to locally analyze data for disruption
+
+ORIGIN_DIR=""
+ALERTS_FILE=""
+DISRUPTION_FILE=""
+DISRUPTION_FILE=""
+GC_CREDS=""
+UPDATE=""
+TEMP_DIR=""
+
+function print_usage() {
+    printf "Usage:
+        -o | Required: root directory location for openshift/origin
+        -a | Required: local alerts file to compare against
+        -d | Required: local disruption file to compare against
+        -g | Optional: google auth file to query ci data from
+        -u | Optional: copy output files back into openshift/origin
+"
+}
+
+function cleanup() {
+    rm -rf $TEMP_DIR
+}
+
+while getopts "o:a:d:g:u" f; do
+    case "$f" in
+    o)
+        ORIGIN_DIR=${OPTARG}
+        ;;
+    a)
+        ALERTS_FILE=${OPTARG}
+        ;;
+    d)
+        DISRUPTION_FILE=${OPTARG}
+        ;;
+    g)
+        GC_CREDS=${OPTARG}
+        ;;
+    u)
+        UPDATE="true"
+        ;;
+    *)
+        print_usage
+        exit 1
+        ;;
+    esac
+done
+
+if [ -z "$ORIGIN_DIR" ]; then
+        print_usage
+        exit 1
+fi
+
+if ([ -z "$ALERTS_FILE" ] || [ -z "$DISRUPTION_FILE" ]) && [ -z "$GC_CREDS" ] ; then
+        print_usage
+        exit 1
+fi
+
+echo "Building job-run-aggregator"
+go build -gcflags='-N -l' `grep "module " go.mod |awk '{print $2}'`/cmd/job-run-aggregator
+
+TEMP_DIR=$(mktemp -d)
+echo "Created temp dir $TEMP_DIR"
+echo "Copying query data"
+cp "$ORIGIN_DIR/pkg/synthetictests/allowedalerts/query_results.json" "$TEMP_DIR/current-alerts.json"
+cp "$ORIGIN_DIR/pkg/synthetictests/allowedbackenddisruption/query_results.json" "$TEMP_DIR/current-disruptions.json"
+
+
+if ! [ -z "$GC_CREDS" ]; then
+    ./job-run-aggregator analyze-historical-data  \
+        --current $TEMP_DIR/current-alerts.json \
+        --data-type alerts \
+        --leeway 1m \
+        --google-service-account-credential-file $GC_CREDS
+
+    ./job-run-aggregator analyze-historical-data  \
+        --current $TEMP_DIR/current-disruptions.json \
+        --data-type disruptions \
+        --leeway 1m \
+        --google-service-account-credential-file $GC_CREDS
+else
+    ./job-run-aggregator analyze-historical-data  \
+        --current $TEMP_DIR/current-alerts.json \
+        --data-type alerts \
+        --new $ALERTS_FILE \
+        --leeway 1m
```

Comment:
Could we move this to a percentage based calculation? A minute is an eternity in disruption world, and if something was 1s and jumped to 50s, we really need someone looking at that. If something grew more than 25-50% we probably should have someone review.

Reply:
Ultimately I'd like something alerting us elsewhere and we could remove the PR review here, right now no such other thing exists.

Reply:
That's a great idea, will update